File: summary/go/6_prime_number.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}

	limit := int(math.Sqrt(float64(n)))
	for i := 2; i <= limit; i++ {
		if n%i == 0 {
			return false
		}
	}

	return true
}

func main() {
	var n int
	fmt.Scan(&n)
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&arr[i])
	}

	for _, v := range arr {
		if isPrime(v) {
			fmt.Printf("%d は素数です。\n", v)
		} else {
			fmt.Printf("%d は素数ではありません。\n", v)
		}
	}
}

File: summary/ts/6_prime_number.ts
function isPrime(num: number): boolean {
  // 0と1は素数ではない
  if (num < 2) {
    return false;
  }

  // 2は最小の素数
  if (num === 2) {
    return true;
  }

  // 偶数は2を除いて素数ではない
  if (num % 2 === 0) {
    return false;
  }

  // 3から sqrt(num) までの奇数で割り切れるかチェック
  const sqrt = Math.sqrt(num);
  for (let i = 3; i <= sqrt; i += 2) {
    if (num % i === 0) {
      return false;
    }
  }

  return true;
}

// 使用例
console.log(isPrime(7)); // true
console.log(isPrime(10)); // false
console.log(isPrime(17)); // true
